src/authentication/__tests__/ExecutionList.test.tsx

```diff
@@ -0,0 +1,150 @@
+import { ExecutionList, IndexChange, LevelChange } from "../execution-model";
+
+describe("ExecutionList", () => {
+  const list2 = new ExecutionList([
+    { id: "1", index: 0, level: 0 },
+    { id: "2", index: 1, level: 0 },
+    { id: "3", index: 0, level: 1 },
+    { id: "4", index: 1, level: 1 },
+    { id: "5", index: 0, level: 2 },
+    { id: "6", index: 1, level: 2 },
+    { id: "7", index: 2, level: 0 },
+  ]);
+
+  it("Move 1 down to the end", () => {
```

Comment:
I prefer writing my test cases like a sentence, for example:

```
it moves 1 down to the end
```

or

```suggestion
  it("moves 1 down to the end", () => {
```

But this is a personal preference :man_shrugging: